src/os/os_freertos.c

```diff
@@ -86,8 +87,9 @@ void oscore_mutex_free(oscore_mutex m) {
 }
 
 int oscore_ncpus(void) {
-	// TODO Use freertos API if available
-	return N_CPUCORES;
+	esp_chip_info_t chip_info;
+	esp_chip_info(&chip_info);
+	return chip_info.cores;
```

Comment:
Oh, hey, that's pretty smart!

We should probably guard this with an ESP32 specific ifdef, or is this something freertos comes with?

Reply:
Yes this is provided by esp32 sdk so a guard would make sense :+1: